### DISCUSSION TITLE
Extend partitionSelect with support for 2D matrices containing a vector

### DISCUSSION THREAD

**Discussion by NexusNull** [josdejong/mathjs#2747] [Ideas]
This is just a suggestion because it strikes me as odd that it doesn't work this way.
Consider this Matrix, it is the result of a subset function.
```json
Matrix {
  "_data": [ [ -150, -100, -0 ] ],
  "_size": [ 1, 3 ],
  "_datatype": "undefined" }
```
Now when calling the partitionSelect function I get an error because the matrix is considered to be two dimensional even when really is only one dimensional. This is nothing major but it could save a couple unnecessary lines of code by converting it to a 1 dimensional matrix.

**Comment by josdejong** [comment]
Thanks for your suggestion. If I understand you correctly you would like  the `partitionSelect` function to accept 2d matrices which contain a vector right?

**Comment by NexusNull** [comment]
Yes

**Comment by josdejong** [comment]
👍  I've marked it as a feature request and changed the title of the issue a bit
